profiling-ffi/src/exporter.rs

```diff
@@ -283,6 +293,25 @@ unsafe fn parse_internal_metadata_json(
     }
 }
 
+unsafe fn parse_system_info_json(
+    system_info_json: Option<&CharSlice>,
+) -> anyhow::Result<Option<serde_json::Value>> {
+    match system_info_json {
+        None => Ok(None),
+        Some(system_info_json) => {
+            let json = system_info_json.try_to_utf8()?;
+            match serde_json::from_str(json) {
+                Ok(parsed) => Ok(Some(parsed)),
+                Err(error) => Err(anyhow::anyhow!(
+                    "Failed to parse contents of system_info json string (`{}`): {}.",
+                    json,
+                    error
+                )),
+            }
+        }
+    }
+}
```

Comment:
Minor: It's maybe worth having a `parse_json` that is used for both system info and internal metadata -- the only practical difference on them seems to be the name of the field ("internal_metadata"/"system_info") on the error string.